components/TokenManagePage/SaveMediaButton.tsx

```diff
@@ -0,0 +1,38 @@
+"use client";
+import { getAddress } from "viem";
+import { useTokenProvider } from "@/providers/TokenProvider";
+import { useUserProvider } from "@/providers/UserProvider";
+import useUpdateTokenURI from "@/hooks/useUpdateTokenURI";
+
+interface SaveMediaButtonProps {
+    title: string;
+    description: string;
+}
+
+const SaveMediaButton = ({ title, description }: SaveMediaButtonProps) => {
+    const { connectedAddress } = useUserProvider();
+    const { owner } = useTokenProvider();
+
+    const { setMetadata, isLoading: isSaving } = useUpdateTokenURI();
+
+    const isOwner = Boolean(
+        connectedAddress && owner &&
+        getAddress(connectedAddress) === getAddress(owner)
+    );

```

Comment:
DRY Principle (don't repeat yourself)
- actual: isOwner logic is defined in both `components/TokenManagePage/SaveMediaButton.tsx` and `components/TokenManagePage/OwnerWarning.tsx`
- required: update `useTokenProvider` to include an export for `isOwner` which you can import into both usages.